src/ejabberd_c2s.erl

```diff
@@ -1258,26 +1259,26 @@ handle_incoming_message({send_text, Text}, StateName, StateData) ->
 handle_incoming_message({broadcast, Accum}, StateName, StateData) ->
     Acc0 = setup_accum(Accum, StateData),
     % DEPRECATED - some obsolete modules broadcast a barebones acc
-    Acc = case mongoose_acc:get(from_jid, Acc0, undefined) of
+    Acc = case mongoose_acc:get_from_jid(Acc0, undefined) of
               undefined ->
                   From = StateData#state.jid,
                   mongoose_acc:update(Acc0, #{from_jid => From,
                       from => jid:to_binary(From)});
               _ -> Acc0
           end,
-    Broadcast = mongoose_acc:get(element, Acc),
+    Broadcast = mongoose_acc:get_element(Acc),
     Acc1 = ejabberd_hooks:run_fold(c2s_loop_debug, Acc, [{broadcast, Broadcast}]),
     ?DEBUG("broadcast=~p", [Broadcast]),
     {Acc2, Res} = handle_routed_broadcast(Acc1, Broadcast, StateData),
     handle_broadcast_result(Acc2, Res, StateName, StateData);
 handle_incoming_message({route, From, To, Acc0}, StateName, StateData) ->
     Acc = setup_accum(Acc0, StateData),
     Acc1 = ejabberd_hooks:run_fold(c2s_loop_debug, Acc, [{route, From, To}]),
-    Name = mongoose_acc:get(name, Acc1),
+    Name = mongoose_acc:get_element_name(Acc1),
     process_incoming_stanza_with_conflict_check(Name, From, To, Acc1, StateName, StateData);
 handle_incoming_message({send_filtered, Feature, From, To, Packet}, StateName, StateData) ->
     % this is used by pubsub and should be rewritten when someone rewrites pubsub module
-    Acc0 = mongoose_acc:new(),
+    Acc0 = ?new_acc(),
```

Comment:
Alternative idea to have descriptive type as an atom `mongoose_acc:new(pubsub_hack_in_c2s)`.
Than we can always grep to get location.